File: Java Backend/UserRegistrationSystem.java
import java.util.HashMap;
import java.util.Map;
import java.util.regex.Pattern;

class User {
    private final String username;
    private final String password;

    public User(String username, String password) {
        this.username = username;
        this.password = password;
    }

    public String getUsername() {
        return username;
    }

    public String getPassword() {
        return password;
    }

    @Override
    public String toString() {
        return "User{" +
                "username='" + username + '\'' +
                ", password='" + password + '\'' +
                '}';
    }
}

public class UserRegistrationSystem {
    private final Map<String, User> users;
    private final Pattern usernamePattern;
    private final Pattern passwordPattern;

    public UserRegistrationSystem() {
        this.users = new HashMap<>();
        this.usernamePattern = Pattern.compile("^[a-zA-Z0-9]+$");
        this.passwordPattern = Pattern.compile("^[a-zA-Z0-9]+$");
    }

    public boolean registerUser(String username, String password) {
        if (!isUsernameValid(username) || !isPasswordValid(password) || isDuplicateUsername(username)) {
            return false;
        }

        User newUser = new User(username, password);
        users.put(username, newUser);
        return true;
    }

    private boolean isUsernameValid(String username) {
        return usernamePattern.matcher(username).matches();
    }

    private boolean isPasswordValid(String password) {
        return passwordPattern.matcher(password).matches();
    }

    private boolean isDuplicateUsername(String username) {
        return users.containsKey(username);
    }

    public static void main(String[] args) {
        // Example usage
        UserRegistrationSystem system = new UserRegistrationSystem();
        System.out.println(system.registerUser("user123", "Password123")); // should return true
        System.out.println(system.registerUser("user123", "password123")); // should return false
        System.out.println(system.registerUser("user1", "pass")); // should return false
    }
}

File: C# Backend/UserRegistrationSystem.cs
using System;
using System.Collections.Generic;
using System.Text.RegularExpressions;

namespace UserRegistrationSystem
{
    public class User
    {
        public string Username { get; }
        public string Password { get; }

        public User(string username, string password)
        {
            Username = username;
            Password = password;
        }

        public override string ToString()
        {
            return $"User: {Username}";
        }
    }

    public class UserRegistrationSystem
    {
        private Dictionary<string, User> users;
        private readonly Regex usernamePattern;
        private readonly Regex passwordPattern;

        public UserRegistrationSystem()
        {
            users = new Dictionary<string, User>();
            usernamePattern = new Regex("^[a-z0-9]{1,}$");
            passwordPattern = new Regex("^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=\\S+$).{8,}$");
        }

        public bool RegisterUser(string username, string password)
        {
            if(!IsUsernameValid(username))
                return false;
            
            var newUser = new User(username, password);
            users.Add(username, newUser);
            return true;
        }

        private bool IsUsernameValid(string username)
        {
            return usernamePattern.IsMatch(username);
        }

        private bool IsPasswordValid(string password)
        {
            return true; // Implement Logic
        }

        private bool IsDuplicateUsername(string username)
        {
            return true; // Implement Logic
        }
        
        static void Main(string[] args)
        {
            
            // Example usage
            var system = new UserRegistrationSystem();
            Console.WriteLine(system.RegisterUser("user123", "Password123")); // should return true
            Console.WriteLine(system.RegisterUser("user123", "password123")); // should return false
            Console.WriteLine(system.RegisterUser("user1", "pass")); // should return false
        }
    }
}
